src/components/player/cue-bar.tsx

```diff
@@ -119,21 +116,21 @@ const NoteCue: React.FC<any> = ({
     // if we seek to the correct time, the note is displayed
     actions.seek(cue.startTime)
     track('opened cue', {cue: cue.text})
+    !muteNotes && setPlayerPrefs({activeSidebarTab: 1})
   }
 
   const close = () => {
     setVisible(false)
   }
 
+  const cueActive = player.activeMetadataTrackCues.includes(cue)
+  const seeking = player.seeking
+  const playerReadyEnough = player.readyState > 0
+
   React.useEffect(() => {
-    if (!muteNotes) {
-      // don't automatically pop if muted
-      setVisible(
-        player.activeMetadataTrackCues.includes(cue) && !player.seeking,
-      )
-      setPlayerPrefs({activeSidebarTab: 1})
-    }
-  }, [player.activeMetadataTrackCues, player.seeking, cue, muteNotes])
+    const isVisible = !muteNotes && cueActive && !seeking && playerReadyEnough
+    setVisible(isVisible)
+  }, [setPlayerPrefs, cueActive, seeking, muteNotes, playerReadyEnough])
```

Comment:
since these are all computer, it makes it more accurate to tun the effect, but I am still getting a brief flash of the previously active note when you switch lessons while it's playing